### PR TITLE
MULE-11298: addition of useConnectorToRetrieveWsdl to manage strategies.

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java [hunk 1]
```diff
@@ -107,20 +105,30 @@ public MuleEvent process(MuleEvent event) throws MuleException
...
 
 
-    private void initializeWsdlRetrievalStrategy()
+    private void initializeWsdlRetrievalStrategy() throws InitialisationException
     {
         HttpRequesterConfig httpRequesterConfig = config.getConnectorConfig();
+        String url = config.getWsdlLocation();
 
-        if (httpRequesterConfig == null || httpRequesterConfig.getProxyConfig() == null)
+        boolean isHttpRequester = url.startsWith(HttpConnector.HTTP);
```
Comment (alepulver) [inline modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java]: This doesn't check the protocol part of the URL: detects http:// but also http.wsdl which is a file.

File: modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java [hunk 2]
```diff
@@ -64,6 +64,8 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.thoughtworks.xstream.InitializationException;
```
Comment (afelisatti) [inline modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java]: This import is wrong.

File: modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java [hunk 3]
```diff
@@ -107,20 +109,49 @@ public MuleEvent process(MuleEvent event) throws MuleException
...
-            wsdlRetrieverStrategy = new ProxyWsdlRetrieverStrategy(httpRequesterConfig.getTlsContext(), httpRequesterConfig.getProxyConfig(), muleContext);
+            wsdlRetrieverStrategy = new URLRetrieverStrategy(url);
         }
     }
 
+    private boolean verifyIsHttpRequester(String url) throws InitialisationException
+    {
+        boolean isHttpRequester = false;
+        URL urlProtocol = null;
+        urlProtocol = IOUtils.getResourceAsUrl(config.getWsdlLocation(), getClass());
```
Comment (afelisatti) [inline modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java]: Is this necessary? Isn't just relevant to check for http and https? Anything else will fall in the URLRetrieverStrategy right?

File: modules/ws/src/main/java/org/mule/module/ws/consumer/WsdlRetrieverStrategy.java [hunk 4]
```diff
@@ -19,5 +17,5 @@
  */
 public interface WsdlRetrieverStrategy 
 {
-	Definition retrieveWsdlFrom(URL url) throws WSDLException;
+	Definition retrieveWsdlFrom() throws WSDLException;
```
Comment (afelisatti) [inline modules/ws/src/main/java/org/mule/module/ws/consumer/WsdlRetrieverStrategy.java]: This method ended up with a wrong name if the URL parameter is gone.

File: modules/ws/src/main/resources/META-INF/mule-ws.xsd [hunk 5]
```diff
@@ -377,6 +377,14 @@
                         </xsd:documentation>
                     </xsd:annotation>
                 </xsd:attribute>
+                <xsd:attribute name="useConnectorToRetrieveWsdl" type="xsd:boolean" use="optional">
```
Comment (afelisatti) [inline modules/ws/src/main/resources/META-INF/mule-ws.xsd]: Where is the default value for this?

File: modules/ws/src/test/java/org/mule/module/ws/consumer/AbstractWSDLServerTestCase.java [hunk 6]
```diff
@@ -25,69 +19,54 @@
...
- * This tests "mocks" a proxy server through which a wsdl file is served.
- *
- */
-public class WSConsumerWsdlProxyTestCase extends FunctionalTestCase
-{
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
+import org.mule.util.IOUtils;
 
+public class AbstractWSDLServerTestCase extends FunctionalTestCase{
```
Comment (afelisatti) [inline modules/ws/src/test/java/org/mule/module/ws/consumer/AbstractWSDLServerTestCase.java]: format, add a space before the bracket

File: modules/ws/src/test/java/org/mule/module/ws/consumer/WSConsumerHttpRequesterTestCase.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+ * The initialization of the ws consumer involved in the test app should
+ * be routed through the http requester. If an URLConnection is used
+ * instead of the http requester, the proxy will not be used. In this
+ * way, we can verify that the http requester is being used.
+ *
+ */
+public class WSConsumerHttpRequesterTestCase extends AbstractWSDLServerTestCase
+{
+	
+	private static final int EXPECTED_NUMBER_OF_WS_CONSUMERS = 1;
```
Comment (afelisatti) [inline modules/ws/src/test/java/org/mule/module/ws/consumer/WSConsumerHttpRequesterTestCase.java]: format, extra tab

File: modules/ws/src/test/resources/ws-consumer-url-connection-test-case.xml [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+	xmlns:http="http://www.mulesoft.org/schema/mule/http" xmlns:vm="http://www.mulesoft.org/schema/mule/vm"
+	xmlns:ws="http://www.mulesoft.org/schema/mule/ws"
+	xsi:schemaLocation="
+        http://www.mulesoft.org/schema/mule/ws http://www.mulesoft.org/schema/mule/ws/current/mule-ws.xsd        
+        http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+        http://www.mulesoft.org/schema/mule/http http://www.mulesoft.org/schema/mule/http/current/mule-http.xsd
+        http://www.mulesoft.org/schema/mule/vm http://www.mulesoft.org/schema/mule/vm/current/mule-vm.xsd">
+
+	<http:request-config name="HTTP_Request_Configuration"
+		protocol="HTTPS" host="fake.com" responseTimeout="60000" port="80"
```
Comment (afelisatti) [inline modules/ws/src/test/resources/ws-consumer-url-connection-test-case.xml]: Is the large timeout necessary?

Comment (afelisatti) [inline modules/ws/src/test/resources/ws-consumer-url-connection-test-case.xml]: Or this will never get called?

Comment (fsgonz) [inline modules/ws/src/test/resources/ws-consumer-url-connection-test-case.xml]: This is never called. Only the initialization is performed as the wsdl retrieval is tested (it is retrieved either from a local file or from a server in localhost via http). However, I removed the timeout attribute.

**Pull request merged**